src/unum/log/log_common.h

```diff
@@ -54,10 +61,12 @@ typedef enum {
 // The 16 lower bits are common flags. The upper 16bits are
 // platform specific flags (LOG_FLAG_FILE, LOG_FLAG_STDOUT...
 // see log_platform.h).
-#define LOG_FLAG_INIT_DONE 0x0001 // The entry has been initialized
-#define LOG_FLAG_INIT_FAIL 0x0002 // The log entry init has failed
-#define LOG_FLAG_MUTEX     0x0004 // The entry requre mutex protextion
-#define LOG_FLAG_INIT_MSG  0x0008 // Log the agent startup info at init
+#define LOG_FLAG_INIT_DONE    0x0001 // The entry has been initialized
+#define LOG_FLAG_INIT_FAIL    0x0002 // The log entry init has failed
+#define LOG_FLAG_MUTEX        0x0004 // The entry requre mutex protextion
+#define LOG_FLAG_INIT_MSG     0x0008 // Log the agent startup info at init
+#define LOG_FLAG_INIT_PER     0x0010 // Log the agent on persistent storage
+#define LOG_FLAG_LOG_CHANGED  0x0020 // Log file changed.
```

Comment:
I don't think those extra flags are necessary. We can't spend time checking for changes on each call so you can just read the prefix on startup, verify path exists and if all ok (if not ok print warning and ignore the prefix) put it into a static storage and set a pointer to that storage in `unum_config` data structure (that way we can also easily add an option to customize the prefix from the command line if needed). By default that pointer in the `unum_config` should point to "/var/log/" constant prefix.

The `lc->name` was meant to be a template, so let's treat it as such, which means (from prefix perspective) we should start changing platform log entries to have only the file name part in that field (i.e. "unum.log", "http.log" e.t.c). In order to have flexibility, when building the name from `lc->name` check first if the template starts with '/' and do not add the prefix then. If it starts with anything else add the prefix and make sure there's '/' between prefix and the name. 

Note: this do require platform repo changes, but cleaner than adding prefix all the time and trying to make subdirs e.t.c..

There's no need to add `log_file[]` field to the log data structure. The names are built only on init and when files are rotated, so it can be done on the fly.

Reply:
@dob71 
Thanks. This seems to be a lighter approach. However as you also said, with the proposed approach, the agent has to wait until the file rotation or agent restart to switch to persistent (or non-persistent).

I chose the current approach to make the agent choose persistent vs non-persistent approach  without restarting the agent. Now that we want the log file to be changed only at the agent start or  on rotation, I have made changes accordingly. Even for this , I have managed to avoid platform specific changes.  And avoid unum_config too. As originally planned, the agent reads the prefix from unum_prefix_file.txt on persistent directory and prepends them to platform specific log file path.
Please let me know if this sounds good.  Or if I am missing anything?

Reply:
Along with efficiency, I am also trying to make it platform independent and avoid platform changes. Please let me know if you  see any issues.